runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientSaslHandshaker.java

```diff
@@ -430,8 +568,98 @@ protected final int decodeSaslAuthenticate(
+                    }
+                    else
+                    {
+                        client.onDecodeSaslResponse(traceId);
+                        client.onDecodeSaslAuthenticateResponse(traceId, authorization, 58);
+                    }
+                }
+                else
+                {
+                    client.onDecodeSaslResponse(traceId);
+                    client.onDecodeSaslAuthenticateResponse(traceId, authorization, errorCode);
+                }
+            }
+        }
+
+        return progress;
+    }
+
+    protected final int decodeSaslScramAuthenticate(
+            KafkaSaslClient client,
+            long traceId,
+            long authorization,
+            long budgetId,
+            int reserved,
+            DirectBuffer buffer,
+            int offset,
+            int progress,
+            int limit)
+    {
+        final int length = limit - progress;
+
+        if (length != 0)
+        {
+            final SaslAuthenticateResponseFW authenticateResponse = saslAuthenticateResponseRO.tryWrap(buffer, progress, limit);
+
+            if (authenticateResponse != null)
+            {
+                final int errorCode = authenticateResponse.errorCode();
+
+                progress = authenticateResponse.limit();
+
+                client.decodableResponseBytes -= authenticateResponse.sizeof();
+                assert client.decodableResponseBytes >= 0;
+
+                try
+                {
+                    byte[] serverKey = formatter.serverKey(client.saltedPassword);
+                    byte[] serverSignature = formatter.hmac(serverKey, client.authMessage);
+                    DirectBuffer serverFinalResponse = authenticateResponse.authBytes().value();
+                    String serverFinalMessage = serverFinalResponse.getStringWithoutLengthUtf8(0,
+                            serverFinalResponse.capacity());
+
+                    Matcher serverFinalResponseMatcher = SERVER_FINAL_MESSAGE.matcher(serverFinalMessage);
+                    while (serverFinalResponseMatcher.find())
+                    {
+                        serverFinalMessage = serverFinalResponseMatcher.group(1);
+                    }
+                    if (!Arrays.equals(Base64.getDecoder().decode(serverFinalMessage),
+                            serverSignature))
+                    {
+                        client.onDecodeSaslResponse(traceId);
+                        client.onDecodeSaslAuthenticateResponse(traceId, authorization, 58);
+                    }
+                    else
+                    {
+                        client.onDecodeSaslResponse(traceId);
+                        client.onDecodeSaslAuthenticateResponse(traceId, authorization, errorCode);
+                    }
+
+                }
+                catch (InvalidKeyException e)
+                {
+                    e.printStackTrace();
+                }
```

Comment:
Avoid these kinds of catch-all exception handlers as they will break the logic of the code when they occur.
Instead determine if the exception is really possible, by inspecting how you generated the inputs, say via `serverKey` and `saltedPassword` above, then remove the `throws` from `hmac(...)` method and use `LangUtil.rethrowUnchecked(ex)` to allow the exception out if it did occur, though you will have already proven to yourself that it is actually not possible.